test/benchmarks/material/button-harness/BUILD.bazel

```
@@ -0,0 +1,82 @@
+load("@npm_angular_dev_infra_private//benchmark/component_benchmark:component_benchmark.bzl", "component_benchmark")
+load("//tools:defaults.bzl", "ng_test_library", "ng_web_test_suite", "ts_library")
+
+package(default_visibility = ["//visibility:public"])
+
+# These are two separate files that provide the same tidy interface for running a performance
+# benchmark.
+#
+# I am not sure why the testbed version of the test does not log console.time to the terminal,
+# although I assume it has something to do with the default karma config that is generated by
+# karma_web_test_suite (which is wrapped by ng_web_test_suite).
+#
+# I am equally lost as to why the performance api exists for the testbed tests but not for the
+# protractor tests.
+#
+# If you have any idea why these things happen, please let me know - @wagnermaciel.
+
+ts_library(
+    name = "constants",
+    srcs = [":constants.ts"],
+)
+
+ts_library(
+    name = "protractor-benchmark-utilities",
+    srcs = [":protractor-benchmark-utilities.ts"],
+    deps = [
+        ":constants",
+        "@npm//@angular/dev-infra-private",
+    ],
+)
+
+ts_library(
+    name = "testbed-benchmark-utilities",
+    srcs = [":testbed-benchmark-utilities.ts"],
+    deps = [":constants"],
+)
+
+# ProtractorHarnessEnvironment
+
+component_benchmark(
+    name = "benchmark",
+    driver = ":protractor.perf-spec.ts",
+    driver_deps = [
+        ":constants",
+        ":protractor-benchmark-utilities",
+        "@npm//protractor",
+        "@npm//@types/jasmine",
+        "//src/cdk/testing",
+        "//src/material/button/testing",
+        "//src/cdk/testing/protractor",
+    ],
+    ng_deps = [
+        ":constants",
+        "@npm//@angular/core",
+        "@npm//@angular/platform-browser",
+        "//src/material/button",
+    ],
+    ng_srcs = [":app.module.ts"],
+    prefix = "",
+    styles = ["//src/material/prebuilt-themes:indigo-pink"],
+)
+
+# TestbedHarnessEnvironment
+
+ng_test_library(
+    name = "unit-test-lib",
+    srcs = ["testbed.perf-spec.ts"],
+    deps = [
+        ":constants",
+        ":testbed-benchmark-utilities",
+        "//src/cdk/testing",
+        "//src/cdk/testing/testbed",
+        "//src/material/button",
+        "//src/material/button/testing",
+        "@npm//@angular/dev-infra-private",
+    ],
+)
+
+ng_web_test_suite(
+    name = "unit-tests",
```

For consistency with other tests we have:

```suggestion
    name = "unit_tests",
```

Alternatively, name it `unit_benchmark_tests` as this is obviously not an actual test, but just a benchmark leveraging the testing browser infrastructure 😄